src/components/Hooks/fetchSortedContacts.js

```diff
@@ -0,0 +1,51 @@
+import { useState } from 'react'
+
+import useFetchContacts from './useFetchContacts'
+import { harmonizeAndSortByFamilyNameGivenNameEmailCozyUrl } from '../../helpers/contacts'
+import {
+  contactsByFamilyNameGivenNameEmailCozyUrl,
+  contactsWithoutIndexes
+} from '../../helpers/queries'
+
+/**
+ * Fetches and harmonize contacts
+ * @param {bool} hasServiceBeenLaunched - keepIndexFullNameAndDisplayNameUpToDate service launch status
+ * @returns {array} [hasContactsToShow, contacts]
+ */
+const fetchSortedContacts = hasServiceBeenLaunched => {
+  const [isContactsWithIndexesFinished, contactsWithIndexes] = useFetchContacts(
+    contactsByFamilyNameGivenNameEmailCozyUrl
+  )
+  const [
+    isContactsWithNoIndexesFinished,
+    contactsWithNoIndexes
+  ] = useFetchContacts(contactsWithoutIndexes)
+
+  const [hasAlreadySortedContacts, setHasAlreadySortedContacts] = useState(
+    false
+  )
+  const [sortedContacts, setSortedContacts] = useState([])
+
+  const fetchesAreFinished =
+    isContactsWithIndexesFinished && isContactsWithNoIndexesFinished
+
+  if (!fetchesAreFinished) {
+    return [false, sortedContacts]
+  }
+
+  if (fetchesAreFinished && !hasAlreadySortedContacts) {
+    const reworkedContacts = hasServiceBeenLaunched
+      ? contactsWithIndexes.concat(contactsWithNoIndexes)
+      : harmonizeAndSortByFamilyNameGivenNameEmailCozyUrl(
+          contactsWithIndexes,
+          contactsWithNoIndexes
+        )
+
+    setHasAlreadySortedContacts(true)
+    setSortedContacts(reworkedContacts)
+  }
+
+  return [true, sortedContacts]
```

Comment:
If we return a static value here, why return it at all?

Reply:
it returned at false in other condition L32 (if !fetchesAreFinished)

Reply:
We can return [fetchesAreFinished, sortedContacts] without any if / else, no?

Reply:
Right, I'll keep only the first one `if (fetchesAreFinished && !hasAlreadySortedContacts)`